Code/Light.GuardClauses/Range.cs

```diff
@@ -214,6 +214,57 @@ public static class Range
     /// <param name="value">The value that indicates the exclusive lower boundary of the resulting range.</param>
     /// <returns>A value you can use to fluently define the upper boundary of a new range.</returns>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static Range<T>.RangeFromInfo FromExclusive<T>(T value) where T : IComparable<T> =>
-        new Range<T>.RangeFromInfo(value, false); 
+    public static Range<T>.RangeFromInfo FromExclusive<T>(T value) where T : IComparable<T> => new (value, false);
+
+    /// <summary>
+    /// Creates a range for the specified enumerable that encompasses all valid indexes.
+    /// </summary>
+    /// <param name="enumerable">
+    /// The count of this enumerable will be used to create the index range. Please ensure that this enumerable
+    /// is actually a collection, not a lazy enumerable.
+    /// </param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="enumerable"/> is null.</exception>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Range<int> For(IEnumerable enumerable)  =>
+        new (0, enumerable.Count(), isFromInclusive: true, isToInclusive: false);
```

Comment:
We should pass `nameof(enumerable)` to the `Count` extension method, so the `ArgumentNullException` will reference the correct parameter name.

Currently the exception will reference the correct `enumerable` parameter only by coincidence, because `EnumerableExtensions.DetermineCountViaEnumerating` two calls deeper also has a parameter named `enumerable`.

Reply:
I've added a dedicated unit test ensuring that "enumerable" is the parameter name when null is passed.
See commit d20736c400556cbf174f7c195b08c69cc3ab6540